### PR TITLE
[JIT] ARM64 - Fixed regressions for `GT_NEG` containment

### LINKED ISSUE #85225: JIT: Invalid result with negated op in compare on arm64
```csharp
// Generated by Fuzzlyn v1.5 on 2023-04-23 16:11:05
// Run on Arm64 Linux
// Seed: 17525435962662079854
// Reduced from 204.8 KiB to 0.2 KiB in 00:02:20
// Debug: Prints 0 line(s)
// Release: Prints 1 line(s)
public class Program
{
    public static uint s_2;
    public static void Main()
    {
        int vr0 = default(int);
        if (56058 < (uint)(-s_2))
        {
            System.Console.WriteLine(vr0);
        }
    }
}
```

cc @TIHan, related to #84667?

### PR DESCRIPTION
**Pull request by TIHan**
Resolves https://github.com/dotnet/runtime/issues/85225

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib @jakobbotsch this is ready

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/codegenarm64.cpp [hunk 1]
```diff
@@ -4575,10 +4575,14 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
                         case GT_RSH:
                         case GT_RSZ:
                         {
+                            if (!op2->gtGetOp1()->isContained())
+                            {
+                                break;
```
Comment (jakobbotsch) [inline src/coreclr/jit/codegenarm64.cpp:4580]: Won't this `break` immediately hit the assert below?

Comment (jakobbotsch) [reply src/coreclr/jit/codegenarm64.cpp:4580]: Ah nevermind, got the sense of the test reversed

**Comment by jakobbotsch** [comment]
Looks like there's still some examples, e.g.:
```csharp
// Generated by Fuzzlyn v1.5 on 2023-04-24 10:09:34
// Run on Arm64 Linux
// Seed: 15668805006347923926
// Reduced from 103.1 KiB to 0.3 KiB in 00:01:18
// Debug: Outputs False
// Release: Outputs True
public class Program
{
    public static long s_7;
    public static int[] s_11 = new int[]{0};
    public static void Main()
    {
        s_11[0] = -2147483648;
        var vr9 = (int)s_7 < (-s_11[0]);
        System.Console.WriteLine(vr9);
    }
}
```

**Comment by TIHan** [comment]
Updated again, this should fix it for good now. I was containing `NEG` for all comparison ops, but it's only valid for `EQ` and `NE`.

**Review by BruceForstall — APPROVED** [review]

**Comment by TIHan** [comment]
CI failure is unrelated. Last Fuzzlyn run succeeded, though it did catch an unrelated case in OSX.

**Pull request merged**